migen/build/generic_platform.py

```diff
@@ -358,3 +371,21 @@ def build(self, fragment):
 
     def create_programmer(self):
         raise NotImplementedError
+
+# makes potentially absolute `path` local to `subdir`, possibly
+# removing a python path to improve legibility
+def _make_local_path(subdir, path):
+    path_parts = []
+    while path != os.path.sep:
```

Comment:
Under what condition is path equal to ``os.path.sep``? ``os.path.split`` removes the path separators.
Also IIRC Windows in some situations accepts both ``/`` and ``\`` as path separator, so a simple comparison against ``os.path.sep`` can cause issues (@cr1901).
Maybe this whole function could make use of Python path objects (``pathlib``).
There's also the special case of the Windows drive letter.

Reply:
>Also IIRC Windows in some situations accepts both / and \ as path separator

Internally, Windows doesn't care what path separator you use. Some software chokes on paths with `/` as a path separator anyway for a myriad of reasons; Xilinx ISE is one of these IIRC.